### ISSUE TITLE
Varnish6

### ISSUE THREAD

**Issue by michbsd** [comotion/VSF#50]
Any chance this could be updated to work with Varnish6 ? 

```
/bin/sh ../libtool  --tag=CC   --mode=compile gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I..   -I/usr/local/include -I/usr/local/include/varnish -I../utf8proc -g -O2 -MT libvmod_vsf_la-vmod_vsf.lo -MD -MP -MF .deps/libvmod_vsf_la-vmod_vsf.Tpo -c -o libvmod_vsf_la-vmod_vsf.lo `test -f 'vmod_vsf.c' || echo './'`vmod_vsf.c
libtool: compile:  gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I.. -I/usr/local/include -I/usr/local/include/varnish -I../utf8proc -g -O2 -MT libvmod_vsf_la-vmod_vsf.lo -MD -MP -MF .deps/libvmod_vsf_la-vmod_vsf.Tpo -c vmod_vsf.c  -fPIC -DPIC -o .libs/libvmod_vsf_la-vmod_vsf.o
In file included from vmod_vsf.c:35:
/usr/local/include/varnish/vrt.h:35:4: error: #error "vrt.h included after cache.h - they are inclusive"
/usr/local/include/varnish/vrt.h:39:4: error: #error "vrt.h included multiple times"
In file included from vmod_vsf.c:35:
/usr/local/include/varnish/vrt.h:127: error: redefinition of 'struct strands'
/usr/local/include/varnish/vrt.h:138: error: redefinition of typedef 'VCL_ACL'
/usr/local/include/varnish/vrt.h:138: error: previous declaration of 'VCL_ACL' was here
/usr/local/include/varnish/vrt.h:139: error: redefinition of typedef 'VCL_BACKEND'
/usr/local/include/varnish/vrt.h:139: error: previous declaration of 'VCL_BACKEND' was here
/usr/local/include/varnish/vrt.h:140: error: redefinition of typedef 'VCL_BLOB'
/usr/local/include/varnish/vrt.h:140: error: previous declaration of 'VCL_BLOB' was here
/usr/local/include/varnish/vrt.h:141: error: redefinition of typedef 'VCL_BODY'
/usr/local/include/varnish/vrt.h:141: error: previous declaration of 'VCL_BODY' was here
/usr/local/include/varnish/vrt.h:142: error: redefinition of typedef 'VCL_BOOL'
/usr/local/include/varnish/vrt.h:142: error: previous declaration of 'VCL_BOOL' was here
/usr/local/include/varnish/vrt.h:143: error: redefinition of typedef 'VCL_BYTES'
/usr/local/include/varnish/vrt.h:143: error: previous declaration of 'VCL_BYTES' was here
/usr/local/include/varnish/vrt.h:144: error: redefinition of typedef 'VCL_DURATION'
/usr/local/include/varnish/vrt.h:144: error: previous declaration of 'VCL_DURATION' was here
/usr/local/include/varnish/vrt.h:145: error: redefinition of typedef 'VCL_ENUM'
/usr/local/include/varnish/vrt.h:145: error: previous declaration of 'VCL_ENUM' was here
/usr/local/include/varnish/vrt.h:146: error: redefinition of typedef 'VCL_HEADER'
/usr/local/include/varnish/vrt.h:146: error: previous declaration of 'VCL_HEADER' was here
/usr/local/include/varnish/vrt.h:147: error: redefinition of typedef 'VCL_HTTP'
/usr/local/include/varnish/vrt.h:147: error: previous declaration of 'VCL_HTTP' was here
/usr/local/include/varnish/vrt.h:148: error: redefinition of typedef 'VCL_INSTANCE'
/usr/local/include/varnish/vrt.h:148: error: previous declaration of 'VCL_INSTANCE' was here
/usr/local/include/varnish/vrt.h:149: error: redefinition of typedef 'VCL_INT'
/usr/local/include/varnish/vrt.h:149: error: previous declaration of 'VCL_INT' was here
/usr/local/include/varnish/vrt.h:150: error: redefinition of typedef 'VCL_IP'
/usr/local/include/varnish/vrt.h:150: error: previous declaration of 'VCL_IP' was here
/usr/local/include/varnish/vrt.h:151: error: redefinition of typedef 'VCL_PROBE'
/usr/local/include/varnish/vrt.h:151: error: previous declaration of 'VCL_PROBE' was here
/usr/local/include/varnish/vrt.h:152: error: redefinition of typedef 'VCL_REAL'
/usr/local/include/varnish/vrt.h:152: error: previous declaration of 'VCL_REAL' was here
/usr/local/include/varnish/vrt.h:153: error: redefinition of typedef 'VCL_STEVEDORE'
/usr/local/include/varnish/vrt.h:153: error: previous declaration of 'VCL_STEVEDORE' was here
/usr/local/include/varnish/vrt.h:154: error: redefinition of typedef 'VCL_STRANDS'
/usr/local/include/varnish/vrt.h:154: error: previous declaration of 'VCL_STRANDS' was here
/usr/local/include/varnish/vrt.h:155: error: redefinition of typedef 'VCL_STRING'
/usr/local/include/varnish/vrt.h:155: error: previous declaration of 'VCL_STRING' was here
/usr/local/include/varnish/vrt.h:156: error: redefinition of typedef 'VCL_TIME'
/usr/local/include/varnish/vrt.h:156: error: previous declaration of 'VCL_TIME' was here
/usr/local/include/varnish/vrt.h:157: error: redefinition of typedef 'VCL_VCL'
/usr/local/include/varnish/vrt.h:157: error: previous declaration of 'VCL_VCL' was here
/usr/local/include/varnish/vrt.h:158: error: redefinition of typedef 'VCL_VOID'
/usr/local/include/varnish/vrt.h:158: error: previous declaration of 'VCL_VOID' was here
/usr/local/include/varnish/vrt.h:160: error: redefinition of 'struct vrt_type'
/usr/local/include/varnish/vrt.h:174: error: redefinition of 'struct vrt_ctx'
/usr/local/include/varnish/vrt.h:215: error: redefinition of 'struct vmod_data'
/usr/local/include/varnish/vrt.h:233: error: nested redefinition of 'enum vcl_event_e'
/usr/local/include/varnish/vrt.h:233: error: redeclaration of 'enum vcl_event_e'
/usr/local/include/varnish/vrt.h:234: error: redeclaration of enumerator 'VCL_EVENT_LOAD'
/usr/local/include/varnish/vrt.h:234: error: previous definition of 'VCL_EVENT_LOAD' was here
/usr/local/include/varnish/vrt.h:235: error: redeclaration of enumerator 'VCL_EVENT_WARM'
/usr/local/include/varnish/vrt.h:235: error: previous definition of 'VCL_EVENT_WARM' was here
/usr/local/include/varnish/vrt.h:236: error: redeclaration of enumerator 'VCL_EVENT_COLD'
/usr/local/include/varnish/vrt.h:236: error: previous definition of 'VCL_EVENT_COLD' was here
/usr/local/include/varnish/vrt.h:237: error: redeclaration of enumerator 'VCL_EVENT_DISCARD'
/usr/local/include/varnish/vrt.h:237: error: previous definition of 'VCL_EVENT_DISCARD' was here
/usr/local/include/varnish/vrt.h:283: error: redefinition of 'struct vrt_backend'
/usr/local/include/varnish/vrt.h:310: error: redefinition of 'struct vrt_backend_probe'
/usr/local/include/varnish/vrt.h:322: error: redefinition of 'struct vrt_ref'
/usr/local/include/varnish/vrt.h:336: error: redefinition of typedef 'acl_match_f'
/usr/local/include/varnish/vrt.h:336: error: previous declaration of 'acl_match_f' was here
/usr/local/include/varnish/vrt.h:338: error: redefinition of 'struct vrt_acl'
/usr/local/include/varnish/vrt.h:359: error: nested redefinition of 'enum gethdr_e'
/usr/local/include/varnish/vrt.h:359: error: redeclaration of 'enum gethdr_e'
/usr/local/include/varnish/vrt.h:360: error: redeclaration of enumerator 'HDR_REQ'
/usr/local/include/varnish/vrt.h:360: error: previous definition of 'HDR_REQ' was here
/usr/local/include/varnish/vrt.h:361: error: redeclaration of enumerator 'HDR_REQ_TOP'
/usr/local/include/varnish/vrt.h:361: error: previous definition of 'HDR_REQ_TOP' was here
/usr/local/include/varnish/vrt.h:362: error: redeclaration of enumerator 'HDR_RESP'
/usr/local/include/varnish/vrt.h:362: error: previous definition of 'HDR_RESP' was here
/usr/local/include/varnish/vrt.h:363: error: redeclaration of enumerator 'HDR_OBJ'
/usr/local/include/varnish/vrt.h:363: error: previous definition of 'HDR_OBJ' was here
/usr/local/include/varnish/vrt.h:364: error: redeclaration of enumerator 'HDR_BEREQ'
/usr/local/include/varnish/vrt.h:364: error: previous definition of 'HDR_BEREQ' was here
/usr/local/include/varnish/vrt.h:366: error: redeclaration of enumerator 'HDR_BERESP'
/usr/local/include/varnish/vrt.h:366: error: previous definition of 'HDR_BERESP' was here
/usr/local/include/varnish/vrt.h:368: error: redefinition of 'struct gethdr_s'
/usr/local/include/varnish/vrt.h:434: error: redefinition of typedef 'vmod_event_f'
/usr/local/include/varnish/vrt.h:434: error: previous declaration of 'vmod_event_f' was here
/usr/local/include/varnish/vrt.h:436: error: redefinition of typedef 'vmod_priv_free_f'
/usr/local/include/varnish/vrt.h:436: error: previous declaration of 'vmod_priv_free_f' was here
/usr/local/include/varnish/vrt.h:437: error: redefinition of 'struct vmod_priv'
In file included from vmod_vsf.c:37:
/usr/local/include/varnish/vsa.h:44: error: expected declaration specifiers or '...' before 'socklen_t'
vmod_vsf.c: In function 'vsf_iter_req_body':
vmod_vsf.c:98: warning: implicit declaration of function 'VSB_bcat'
vmod_vsf.c: In function '__match_proto__':
vmod_vsf.c:104: error: expected declaration specifiers before 'vmod_body'
vmod_vsf.c:149: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'vmod_conn_reset'
vmod_vsf.c:168: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'vmod_urldecode'
vmod_vsf.c:192: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'vmod_normalize'
vmod_vsf.c:235: warning: type of 'td_vsf_body' defaults to 'int'
vmod_vsf.c:235: error: expected '{' at end of input
gmake[3]: *** [Makefile:651: libvmod_vsf_la-vmod_vsf.lo] Error 1
gmake[3]: Leaving directory '/root/work/varnish-vmods/VSF/libvmod-vsf/src'
gmake[2]: *** [Makefile:465: all-recursive] Error 1
gmake[2]: Leaving directory '/root/work/varnish-vmods/VSF/libvmod-vsf'
gmake[1]: *** [Makefile:376: all] Error 2
gmake[1]: Leaving directory '/root/work/varnish-vmods/VSF/libvmod-vsf'
gmake: *** [Makefile:12: libvmod-vsf/src/.libs/libvmod-vsf.so] Error 2
```

**Comment by fgsch** [comment]
Sure, let me sort it out and I will update it.

**Comment by michbsd** [comment]
If it helps, I got it compiling with the following patch : 

```
--- vmod_vsf.c.orig	2018-03-29 14:35:21.103649000 +0200
+++ vmod_vsf.c	2018-03-29 15:05:42.293841000 +0200
@@ -26,15 +26,21 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */

+#include "config.h"
+
 #include <ctype.h>
 #include <stdlib.h>
 #include <utf8proc.h>

-#include "cache/cache.h"
+#include "cache/cache_varnishd.h"
+#include <vcl.h>
+
+#ifndef VRT_H_INCLUDED
+#  include <vrt.h>
+#endif

-#include "vrt.h"
-#include "vcl.h"
 #include "vsa.h"
+#include "vsb.h"
 #include "vtcp.h"

 #include "vcc_if.h"
@@ -100,7 +106,7 @@
 }
 #endif

-VCL_STRING __match_proto__(td_vsf_body)
+VCL_STRING v_matchproto_(td_vsf_body)
 vmod_body(VRT_CTX, struct vmod_priv *priv, VCL_BYTES maxsize)
 {
 	struct vsb *vsb;
@@ -145,7 +151,7 @@
  * Copyright (c) 2011 Varnish Software AS
  * All rights reserved.
  */
-VCL_VOID __match_proto__(td_vsf_conn_reset)
+VCL_VOID v_matchproto_(td_vsf_conn_reset)
 vmod_conn_reset(VRT_CTX)
 {
 	CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);
@@ -164,7 +170,7 @@
 	SES_Close(ctx->req->sp, SC_RESP_CLOSE);
 }

-VCL_STRING __match_proto__(td_vsf_urldecode)
+VCL_STRING v_matchproto_(td_vsf_urldecode)
 vmod_urldecode(VRT_CTX, VCL_STRING s)
 {
 	unsigned u, v;
@@ -188,7 +194,7 @@
 	}
 }

-VCL_STRING __match_proto__(td_vsf_normalize)
+VCL_STRING v_matchproto_(td_vsf_normalize)
 vmod_normalize(VRT_CTX, VCL_STRING s)
 {
 	char *p;

```

**Comment by michbsd** [comment]
Will this make it into the release branch?

**Comment by Anasxrt** [comment]
I am waiting for this patch as well 👍

**Comment by michbsd** [comment]
@Anasxrt I've submitted a PR :-)

**Comment by fgsch** [comment]
Apologies for the delay.

I will work on this over the weekend.

**Comment by fgsch** [comment]
Thanks for your patience with this issue.

I've created a 6.0 branch and fixed vsf in d657f638b4e7e6c0d02de7415c59de0be14bfa1a.
I will tackle vsthrottle next.

**Comment by fgsch** [comment]
Fixed in d2070d4bcda22acb3dc58a3211b433de5eea12e4.

If you are still experiencing problems please open a new issue.
